lib/index.js

```diff
@@ -260,23 +219,33 @@ module.exports = {
       var _nugtKeyCount = Object.keys(nuspecResolutions).length;
       Object.keys(flattendPackageList).forEach(function (packageName) {
         tree.dependencies[packageName] =
-          flattendPackageList[packageName].cloneShallow()
-      })
+          flattendPackageList[packageName].cloneShallow();
+      });
       if (_nugtKeyCount > 0) {
         // local folders scanned, build list from .nuspec
         for (var key in nuspecResolutions) {
           var resolution = nuspecResolutions[key];
           var node = flattendPackageList[resolution.name].cloneShallow();
-          node.from = tree.from.concat()
+          node.from = tree.from.concat();
           buildTree(node, resolution.children, flattendPackageList);
           tree.dependencies[node.name] = node;
         }
       }
       return packageTree;
     })['catch'](function (err) {
       throw(err);
-    })
+    });
 
     return job;
   },
-};
\ No newline at end of file
+};
+
+function injectPath(dep, packagesFolder) {
+  dep.path =
+    dep.localPath ?
+    path.resolve(packagesFolder, dep.localPath)
+    : path.resolve(packagesFolder, dep.name + '.' + dep.version);
+  if (dep.localPath) {
+    delete dep.localPath;
+  }
```

Comment:
Why is the deletion needed? This changes `dep` in a way that concurrent calls to this func will lead to different results. Seems 'surprising'?

Reply:
I'll take a look.
btw, I didn't write this function, just refactored it out from here https://github.com/snyk/snyk-nuget-plugin/pull/27/files#diff-6d186b954a58d5bb740f73d84fe39073L155